frontend/src/job/ingest-journal/ZenonImport.ts

```diff
@@ -0,0 +1,64 @@
+import axios from 'axios';
+
+const zenonBaseURL: string = 'https://zenon.dainst.org/api/v1/';
+
+function isId(term: string) {
+    return !!term.match(/^\w?\w?\d{9}$/);
+}
+
+export default async function search(term: string): Promise<ZenonResultData> {
```

Comment:
Hier meckert CodeBeat, die Funktion sei zu lang. Theoretisch könnte man die zwei Schritte von Query bauen und Query abschicken in unterschiedliche Methoden auslagern?

Ein spontaner Gedanke zu record/search: Wieso nicht zwei Funktionen? Eine Suche, und eine "getRecord"? 

Bei der Suche würde ich als Optionen eher erwarten, dass mans ich entscheiden kann in allen Feldern, nach Autoren oder nach Titeln zu suchen - anstatt hardcoded nur nach Titeln.

Reply:
Ja habe gesehen, dass Codebeat meckert, hab mich aber entschieden das zu ignorieren, weil ich die so OK finde. Aber wenn du/ihr meint, kann ich das aufteilen.

Was das Tunen der Suche angeht, gebe ich dir schon recht - das wäre nicht verkehrt. Weiss halt nicht wie da die Anforderungen sind und hab's im Zweifel so übernommen wie es im alten UI war. Dann würde ich es so machen, dass man neben dem Eingabefeld noch eine Combobox hat und auswählen kann wo man suchen will: Alle Felder, Titel, Autoren. Oder noch weitere?

Reply:
Ich würde erstmal die 3 nehmen.